### ISSUE TITLE
StartupCostTimesUtils LinkedHashMap crashes

### ISSUE THREAD

**Issue by dmitriyTarasov** [idisfkj/android-startup#16]
Hi,
`StartupCostTimesUtils` uses `LinkedHashMap` for storing `CostTimesModel`, but `StartupCostTimesUtils#recordStart` is called from different threads. In some cases there were crashes. 

Can you please add the ability to disable statistics for release builds?

Thanks!

Crashes:

```
Fatal Exception: java.lang.ArrayIndexOutOfBoundsException: length=16; index=28
       at java.util.LinkedHashMap.addNewEntry(LinkedHashMap.java:197)
       at java.util.HashMap.put(HashMap.java:403)
       at com.rousetime.android_startup.utils.StartupCostTimesUtils.recordStart(StartupCostTimesUtils.java:24)
       at com.rousetime.android_startup.run.StartupRunnable.run(StartupRunnable.java:31)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
       at java.lang.Thread.run(Thread.java:818)
```

For Android 7 (maybe a bug in android implementation of LinkedHashMap):
```
Fatal Exception: java.lang.NullPointerException: Attempt to read from field 'int java.util.HashMap$HashMapEntry.hash' on a null object reference
       at java.util.LinkedHashMap.transfer(LinkedHashMap.java:273)
       at java.util.HashMap.resize(HashMap.java:512)
       at java.util.HashMap.addEntry(HashMap.java:808)
       at java.util.LinkedHashMap.addEntry(LinkedHashMap.java:464)
       at java.util.HashMap.put(HashMap.java:436)
       at com.rousetime.android_startup.utils.StartupCostTimesUtils.recordStart(StartupCostTimesUtils.java:24)
       at com.rousetime.android_startup.run.StartupRunnable.run(StartupRunnable.java:31)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
       at java.lang.Thread.run(Thread.java:762)
```

**Comment by idisfkj** [comment]
Thanks, I will fix it in the next version.

I will solve it in the following way:

1. Use `ConcurrentHashMap` to replace `LinkedHashMap` to solve the problem of multithreading concurrency.
2. Provide the `openStatistic` parameter in `StartupConfig` to set the statistics switch.

**Comment by idisfkj** [comment]
This has been fixed in version [1.0.7](https://github.com/idisfkj/android-startup/releases/tag/1.0.7).
